src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_ISwitchExpression.cs

```diff
@@ -719,6 +719,121 @@ void M(bool result, int input)
             Leaving: {R1}
 }
 
+Block[B8] - Exit
+    Predecessors: [B7]
+    Statements (0)
+";
+            VerifyFlowGraphAndDiagnosticsForTest<BlockSyntax>(source, expectedFlowGraph, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [Fact, WorkItem(32216, "https://github.com/dotnet/roslyn/issues/32216")]
+        public void SwitchExpression_CompoundGuard()
+        {
+            string source = @"
+#pragma warning disable CS8509
+public sealed class MyClass
+{
+    void M(bool result, int input, bool a, bool b)
+    /*<bind>*/{
+        result = input switch
+            {
+                1 when a && b => false
```

Comment:
>1 [](start = 16, length = 1)

Is it possible for the compiler to bind bad code that has flow in this pattern as a switch expression? If so, please add some CFG verification. If not, could you please add a regular IOp test showing what the behavior is? ie, something like `input switch { test1 && test2 => true }` or similar.